### PR TITLE
[server-wallet] Use @statechannels/wasm-utils

### PR DESCRIPTION
**Pull request by andrewgordstewart**
I'd like to see if this works on circle.

### CODE REVIEW

**Comment by geoknee** [comment]
Seeing 
```
@statechannels/server-wallet:     RuntimeError: unreachable
@statechannels/server-wallet:         at wasm-function[243]:0x45426
@statechannels/server-wallet:         at wasm-function[320]:0x478ab
@statechannels/server-wallet:         at wasm-function[337]:0x47c3f
@statechannels/server-wallet:         at wasm-function[279]:0x46a97
@statechannels/server-wallet:         at wasm-function[122]:0x3be4d
@statechannels/server-wallet:         at wasm-function[312]:0x4766d
```

**Comment by geoknee** [comment]
The errors went away when I realised I was passing in `server-wallet.State` instead of `nitro-protocol.State`.

**Comment by geoknee** [comment]
@andrewgordstewart would it make more sense to switch to these utils in `nitro-protocol`, rather than `server-wallet` (or even `wallet-core`)?

**Comment by andrewgordstewart** [comment]
Good idea.

The package also has types, so I'm surprised it didn't catch that, but perhaps it's not properly configured.

**Comment by geoknee** [comment]
> The package also has types, so I'm surprised it didn't catch that, but perhaps it's not properly configured.

Yes, see this [discussion](https://github.com/statechannels/native-utils/pull/4#issuecomment-698298326).

**Comment by geoknee** [comment]
I propose that we tackle converting our entire stack to use this wasm stuff in a separate PR. That should

- make it much easier to avoid accidentally using some slower utils 
- share the speed advantages with other applications (e.g. browser wallet)
- cut down on `thisTypeToThatType` gymnastics

and my rationale for separate PR is that there are some questions:
- will there be a circular dependency relationship between `native-utils` and `nitro-protocol`?
- should `native-utils` actually just be part of `nitro-protocol`?
- will it seamlessly run in the browser?

**Comment by geoknee** [comment]
100 makePayment calls taking 
b4cddc8:  [86948 ms](https://app.circleci.com/pipelines/github/statechannels/statechannels/9184/workflows/f5008bea-d152-4b39-a947-4e3ea2e073e4/jobs/44191)
(master) 849f9b6 : [138799 ms ](https://app.circleci.com/pipelines/github/statechannels/statechannels/9179/workflows/f2eaaecf-0428-4258-a068-42e63357b5d3/jobs/44161)

Which looks to be a pretty nice speedup, notwithstanding issues with trusting these numbers as discussed in #2606

**Comment by geoknee** [comment]
Rebased, and snuck in a version bump for `wasm-utils` (which has better typings).

File: packages/server-wallet/src/utilities/signatures.ts [hunk 1]
```diff
@@ -1,59 +1,25 @@
...
 ): Promise<{state: State; signature: string}> {
   const address = cachedAddress(privateKey);
   if (state.participants.map(p => p.signingAddress).indexOf(address) < 0) {
     throw new Error("The state must be signed with a participant's private key");
   }
 
-  const {stateHash} = state;
-
-  const signature = await fastSignData(stateHash, privateKey);
+  const signature = await wasmUtils.signState(toNitroState(state), privateKey).signature;
```
Comment (andrewgordstewart) [inline packages/server-wallet/src/utilities/signatures.ts:19]: I think this is now synchronous.

With bitauth, we needed to await the EC to be constructed, and thus this needed to be async.

But the wasm-utils are synchronous -- I think it probably constructs a EC on each call, which is suboptimal, but it still "fast enough".

**Review by andrewgordstewart — COMMENTED** [review]
GitHub says I'm not allowed to approve this pr, but I approve it anyway!

**Review by geoknee — APPROVED** [review]
Great work ;-)

File: packages/server-wallet/src/utilities/signatures.ts (hunk 1 above)
Comment (geoknee) [reply packages/server-wallet/src/utilities/signatures.ts:19]: #2619

**Pull request merged**